### PR TITLE
docs: Add more details and examples to SDL and Data Model.

### PR DESCRIPTION
**Pull request by vpetrovykh**

### CODE REVIEW

**Comment by vpetrovykh** [comment]
I've moved the descriptive text inside the abstract type definition bodies in the docs.

File: docs/edgeql/sdl/constraints.rst [hunk 1]
```diff
@@ -7,6 +7,32 @@ Constraints
 This section describes the SDL declarations pertaining to
 :ref:`constraints <ref_datamodel_constraints>`.
 
+
+Examples
+-------_
```
Comment (1st1) [inline docs/edgeql/sdl/constraints.rst]: `-_` -> `--`

File: docs/edgeql/sdl/constraints.rst [hunk 2]
```diff
@@ -7,6 +7,32 @@ Constraints
...
+            'Minimum allowed value for {__subject__} is {min}.';
+        expr := __subject__ >= min;
+    }
+
+Declare a *concrete* constraint on an integer type:
+
+.. code-block:: sdl
+
+    scalar type posint64 extending int64 {
+        constraint min(0);
```
Comment (1st1) [inline docs/edgeql/sdl/constraints.rst]: min_value

File: docs/edgeql/sdl/indexes.rst [hunk 3]
```diff
@@ -7,6 +7,28 @@ Indexes
...
+.. code-block:: sdl
+
+    type User {
+        property name -> str;
+        property address -> str;
+
+        multi link friends -> User;
+
+        # define an index for User based on name
+        index user_name_idx on __subject__.name;
```
Comment (1st1) [inline docs/edgeql/sdl/indexes.rst]: on ()

File: docs/edgeql/sdl/constraints.rst (hunk 1 above)
Comment (vpetrovykh) [reply docs/edgeql/sdl/constraints.rst]: Fixed.

File: docs/edgeql/sdl/constraints.rst (hunk 2 above)
Comment (vpetrovykh) [reply docs/edgeql/sdl/constraints.rst]: OK

File: docs/edgeql/sdl/indexes.rst (hunk 3 above)
Comment (vpetrovykh) [reply docs/edgeql/sdl/indexes.rst]: Fixed

**Comment by vpetrovykh** [comment]
I've also added a note to the individual synopsis descriptions drawing attention to the parentheses in the  ``on ( <expr> )``. Since it's a small and pesky bit of grammar that may be overlooked at a quick glance.

**Pull request merged**